src/app/services/user.ts

```diff
@@ -1,8 +1,9 @@
-import { isDefined } from "./";
+import { USER_ROLES, isDefined } from "./";
 import { LoggedInUser, PotentialUser, School } from "../../IsaacAppTypes";
-import { UserRole } from "../../IsaacApiTypes";
 import { Immutable } from "immer";
 
+export type UserRole = (typeof USER_ROLES)[number];
```

Comment:
Given that `USER_ROLES` contains all the possible roles in `Role`, I think this could mean that `UserRole` is an alias of `Role`. Unless there is a good reason why we're doing this, I would have thought that we should be using `Role` instead (and `UserRole` should be removed). Thoughts?

Reply:
It looks like this is not possible, because you can't convert a union type (which is a compile-time constant) into an array with those strings in it (which is a run-time value).